src/bun.js/bindings/v8/Object.cpp

```diff
@@ -0,0 +1,58 @@
+#include "v8/Object.h"
+
+#include "JavaScriptCore/ObjectConstructor.h"
+
+using JSC::Identifier;
+using JSC::JSFinalObject;
+using JSC::JSGlobalObject;
+using JSC::JSObject;
+using JSC::JSValue;
+using JSC::PutPropertySlot;
+
+namespace v8 {
+
+Local<Object> Object::New(Isolate* isolate)
+{
+    JSFinalObject* object = JSC::constructEmptyObject(isolate->globalObject());
+    JSValue jsv(object);
+    return Local<Object>(jsv);
+}
+
+Maybe<bool> Object::Set(Local<Context> context, Local<Value> key, Local<Value> value)
+{
+    JSGlobalObject* globalObject = *context;
+    JSObject* object = toJSValue().getObject();
+    assert(object);
```

Comment:
Note that `getObject()` does this assertion internally. It will crash at runtime if it's not an object.